File: ts-tests/src/pages/alert-page.ts
import { Page, Locator } from '@playwright/test';

export class AlertPageTs {
  readonly page: Page;

  constructor(page: Page) {
    this.page = page;
  }

  async navigate() {
    await this.page.goto('https://www.qaplayground.com/practice/alert');
  }

  private simpleAlertButton(): Locator {
    return this.page.locator("button:has-text('Simple Alert')").first();
  }

  private confirmAlertButton(): Locator {
    return this.page.locator("button:has-text('Confirm Alert')").first();
  }

  private promptAlertButton(): Locator {
    return this.page.locator("button:has-text('Prompt Alert')").first();
  }

  async clickSimpleAlertAndGetText(): Promise<string> {
    let message = '';
    this.page.once('dialog', dialog => {
      message = dialog.message();
      dialog.accept();
    });
    await this.simpleAlertButton().click();
    await this.page.waitForTimeout(500);
    return message;
  }

  async clickConfirmAlertAndAccept(): Promise<string> {
    let message = '';
    this.page.once('dialog', dialog => {
      message = dialog.message();
      dialog.accept();
    });
    await this.confirmAlertButton().click();
    await this.page.waitForTimeout(500);
    return message;
  }

  async clickConfirmAlertAndDismiss(): Promise<string> {
    let message = '';
    this.page.once('dialog', dialog => {
      message = dialog.message();
      dialog.dismiss();
    });
    await this.confirmAlertButton().click();
    await this.page.waitForTimeout(500);
    return message;
  }

  async clickPromptAlertAndEnterName(name: string): Promise<string> {
    let message = '';
    this.page.once('dialog', dialog => {
      message = dialog.message();
      dialog.accept(name);
    });
    await this.promptAlertButton().click();
    await this.page.waitForTimeout(500);
    return message;
  }
}

File: py-tests/pages/alert_page.py
from playwright.sync_api import Page, Locator


class AlertPage:
    def __init__(self, page: Page) -> None:
        self.page = page

    def navigate(self) -> "AlertPage":
        self.page.goto("https://www.qaplayground.com/practice/alert")
        return self

    def _simple_alert_button(self) -> Locator:
        return self.page.locator("button:has-text('Simple Alert')").first

    def _confirm_alert_button(self) -> Locator:
        return self.page.locator("button:has-text('Confirm Alert')").first

    def _prompt_alert_button(self) -> Locator:
        return self.page.locator("button:has-text('Prompt Alert')").first

    def click_simple_alert_and_get_text(self) -> str:
        message = ""

        def handle_dialog(dialog):
            nonlocal message
            message = dialog.message
            dialog.accept()

        self.page.once("dialog", handle_dialog)
        self._simple_alert_button().click()
        self.page.wait_for_timeout(500)
        return message

    def click_confirm_alert_and_accept(self) -> str:
        message = ""

        def handle_dialog(dialog):
            nonlocal message
            message = dialog.message
            dialog.accept()

        self.page.once("dialog", handle_dialog)
        self._confirm_alert_button().click()
        self.page.wait_for_timeout(500)
        return message

    def click_confirm_alert_and_dismiss(self) -> str:
        message = ""

        def handle_dialog(dialog):
            nonlocal message
            message = dialog.message
            dialog.dismiss()

        self.page.once("dialog", handle_dialog)
        self._confirm_alert_button().click()
        self.page.wait_for_timeout(500)
        return message

    def click_prompt_alert_and_enter_name(self, name: str) -> str:
        message = ""

        def handle_dialog(dialog):
            nonlocal message
            message = dialog.message
            dialog.accept(name)

        self.page.once("dialog", handle_dialog)
        self._prompt_alert_button().click()
        self.page.wait_for_timeout(500)
        return message
